### PR TITLE
<InputWithOptions/> - prevent submit event default behaviour on Enter key

### PR DESCRIPTION
**Pull request by gilmeir**
#2657

### CODE REVIEW

File: src/Layout/docs/examples/Form.js [hunk 1]
```diff
@@ -9,9 +9,49 @@ import Checkbox from 'wix-style-react/Checkbox';
 import Text from 'wix-style-react/Text';
 import Button from 'wix-style-react/Button';
 import RadioGroup from 'wix-style-react/RadioGroup';
-
+import MultiSelect from 'wix-style-react/MultiSelect';
 import styles from '../styles.scss';
 
+class MultiSelectForTest extends React.Component {
```
Comment (shlomitc) [inline src/Layout/docs/examples/Form.js]: why did you need to add this?

File: src/Input/Input.js [hunk 2]
```diff
@@ -405,7 +405,7 @@ class Input extends Component {
     const { customInput: CustomInputComponent, dataRef, ...rest } = props;
     const inputProps = {
       ...(CustomInputComponent
-        ? { 'data-ref': dataRef, ...rest, 'data-hook': 'wsr-custom-input' }
+        ? { 'data-ref': dataRef, ...rest, 'data-hook': 'wsr-input-custom' }
```
Comment (shlomitc) [inline src/Input/Input.js]: why is this change necessary?

File: src/Layout/docs/examples/Form.js (hunk 1 above)
Comment (gilmeir) [reply src/Layout/docs/examples/Form.js]: I wanted this use case (input with options inside form) to be available in a storybook example, so one can easily try out hitting Enter easily.

Are there any constraints about changing this example?

Comment (gilmeir) [reply src/Layout/docs/examples/Form.js]: I wanted to add this use case to the Form example in storybook, so if anyone wants to manually test the behaviour of InputWithOptions inside a form they can open this example.

File: src/Input/Input.js (hunk 2 above)
Comment (gilmeir) [reply src/Input/Input.js]: This allows us to find both custom and not custom Input elements in the protractor inputDriver:
https://github.com/wix/wix-style-react/blob/inputwoptions-enter-inside-form/src/Input/Input.protractor.driver.js#L4

Both data-hooks are defined here in Input:
1. https://github.com/wix/wix-style-react/blob/inputwoptions-enter-inside-form/src/Input/Input.js#L200
2. https://github.com/wix/wix-style-react/blob/inputwoptions-enter-inside-form/src/Input/Input.js#L408

File: src/InputWithOptions/InputWithOptions.protractor.driver.js [hunk 3]
```diff
@@ -1,21 +1,24 @@
...
-    isFocused: () => isFocused(input),
+    click: () => inputDriver.click(),
+    getInput: () => inputDriver.element(),
+    getInputText: () => inputDriver.getText(),
+    isFocused: () => inputDriver.isFocused(),
     element: () => component,
     /** Check wether the options dropdown is open */
     isOptionsShown: () => dropdownLayoutDriver.getDropdown().isDisplayed(),
-    enterText: text => input.clear().sendKeys(text),
+    enterText: text => inputDriver.enterText(text),
```
Comment (shlomitc) [inline src/InputWithOptions/InputWithOptions.protractor.driver.js]: might be a breaking change, let's revery this line

File: src/InputWithOptions/InputWithOptions.js [hunk 4]
```diff
@@ -171,7 +171,11 @@ class InputWithOptions extends WixComponent {
     return (
       <div>
         {dropDirectionUp ? this._renderDropdownLayout() : null}
-        <div data-input-parent className={this.inputClasses()}>
+        <div
+          data-input-parent
+          className={this.inputClasses()}
+          data-hook="input-with-options--input"
```
Comment (shlomitc) [inline src/InputWithOptions/InputWithOptions.js]: this doesn't looks like the right place to put the data-hook.
if you want to use the `inputDriver`, then you should mark the `<Input/>` itself with this data hook

File: src/Layout/docs/examples/Form.js (hunk 1 above)
Comment (shlomitc) [reply src/Layout/docs/examples/Form.js]: it adds complexity that isn't needed for this example, let's remove it

File: src/Input/Input.protractor.driver.js [hunk 5]
```diff
@@ -1,8 +1,9 @@
 import { isFocused } from 'wix-ui-test-utils/protractor';
 
 const inputDriverFactory = component => {
-  const input = component.$('[data-hook="wsr-input"]'),
-    clearButton = component.$('[data-hook="input-clear-button"]');
+  // selector doesn't use data-hook since Input consumer can provide his own customInput
+  const input = component.$('input');
```
Comment (shlomitc) [inline src/Input/Input.protractor.driver.js]: see the explanation regarding `input-with-options--input`

File: src/Input/Input.js (hunk 2 above)
Comment (gilmeir) [reply src/Input/Input.js]: reverting this change
